lib/amqp/gen/consumer.ex

```diff
@@ -197,8 +198,17 @@ defmodule Amqpx.Gen.Consumer do
     e in _ ->
       Logger.error(inspect(e))
 
+      is_message_to_reject =
+        function_exported?(handler_module, :handle_message_rejection, 1) &&
+          redelivered
+
       Task.start(fn ->
         :timer.sleep(backoff)
+
+        if is_message_to_reject do
```

Comment:
```suggestion
      Task.start(fn ->
        :timer.sleep(backoff)

        if function_exported?(handler_module, :handle_message_rejection, 1) && redelivered do
```
or define it in a defp.